edk2toolext/versioninfo/versioninfo_helper.py

```diff
@@ -0,0 +1,556 @@
+            valid = False
+
+        if self.versionDict[FILE_TYPE_STR] not in VALID_FILE_TYPE_VALUES:
+            print("ERROR: Invalid FILETYPE value: " + self.versionDict[FILE_TYPE_STR] + ".", file=sys.stderr)
+            valid = False
+
+        if self.versionDict[FILE_TYPE_STR] == "VFT_DRV":
+            if self.versionDict[FILE_SUBTYPE_STR] not in VALID_SUBTYPE_VFT_DRV:
+                print("ERROR: Invalid FILESUBTYPE value for FILETYPE VFT_DRV: "
+                      + self.versionDict[FILE_SUBTYPE_STR] + ".", file=sys.stderr)
+                valid = False
+        elif self.versionDict[FILE_TYPE_STR] == "VFT_FONT":
+            if self.versionDict[FILE_SUBTYPE_STR] not in VALID_SUBTYPE_VFT_FONT:
+                print("ERROR: Invalid FILESUBTYPE value for FILETYPE VFT_FONT: "
+                      + self.versionDict[FILE_SUBTYPE_STR] + ".", file=sys.stderr)
+                valid = False
+        elif self.versionDict[FILE_TYPE_STR] != "VFT_VXD" and self.versionDict[FILE_SUBTYPE_STR] != 0:
+            print("ERROR: Invalid FILESUBTYPE value for FILETYPE "
+                  + self.versionDict[FILE_TYPE_STR] + ", value must be 0.", file=sys.stderr)
+            valid = False
+
+        if self.__validateVersionNumber__(self.versionDict[STRING_FILE_INFO_STR][FILE_VERSION_STR]):
+            if self.versionDict[STRING_FILE_INFO_STR][FILE_VERSION_STR] != self.versionDict[FILE_VERSION_STR]:
+                print("ERROR: FILEVERSION in header does not match FileVersion in StringFileInfo.", file=sys.stderr)
+                valid = False
+        else:
+            valid = False
+
+        if TRANSLATION_STR in self.versionDict[VAR_FILE_INFO_STR]:
+            langIDset = self.versionDict[VAR_FILE_INFO_STR][TRANSLATION_STR].split(" ")
+            try:
+                if len(langIDset) != 2:
+                    print("ERROR: Translation field must contain 2 space delimited hexidecimal 8 bit words.",
+                          file=sys.stderr)
+                    valid = False
+                elif int(langIDset[0], 0) not in VALID_LANG_ID:
+                    print("ERROR: Invalid language code: " + langIDset[0] + "\".", file=sys.stderr)
+                    valid = False
+                elif int(langIDset[1], 0) not in VALID_CHARSET_ID:
+                    print("ERROR: Invalid charset code: " + langIDset[1] + "\".", file=sys.stderr)
+                    valid = False
+            except ValueError:
+                print("ERROR: Invalid language code: " + langIDset[0] + "\".", file=sys.stderr)
+                valid = False
+        else:
+            print("ERROR: Missing required parameter: Translation in VarFileInfo", file=sys.stderr)
+            valid = False
+
+        return valid
+
+    def write(self, path):
+        if not self.validate():
+            return
+
+        outStr = "/* Auto-generated VERSIONINFO resource file. */\n\n"
```

Comment:
might be good to add timestamp as well as tool version that way debugging a tool error is easier at a later point in time when there are multiple versions of the tool